### PR TITLE
Update API docs for v5

### PR DESCRIPTION
**Pull request by jimbolla**
This can certainly be better, but it's not a bad start.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -294,3 +291,80 @@ function mergeProps(stateProps, dispatchProps, ownProps) {
...
+import * as actionCreators from './actionCreators'
+import { bindActionCreators } from 'redux'
+
+function selectorFactory(dispatch) {
+  let state = {}
+  let ownProps = {}
+  let result = {}
+  const actions = bindActionCreators(actionCreators, dispatch)
+  const addTodo = (text) => actions.addTodo(ownProps.userId, text)
+  const memoize = initShallowMemoizer()
```
Comment (slorber) [inline docs/api.md]: Hey, not sure to have seen this memoizer used in this example

Comment (jimbolla) [inline docs/api.md]: Whoops. It was supposed to be part of the example. Must've got lost. I'll update.

**Comment by timdorr** [comment]
OK, let's merge this in for now.

**Comment by mondaychen** [comment]
Hi.

I was reading https://github.com/reactjs/react-redux/blob/master/docs/api.md#examples-1 and this PR contains the latest commit changing that part.

I guess you have a plan to update the example for `connectAdvanced`, but it was not done yet. That's why it now confuses me.

- It seems that `nextProps` should be `nextOwnProps`. Correct? Otherwise it would be an undefined variable.
- What's the use of `state`? It keeps track of the newest state, but is not used to compute anything in the scope.

**Pull request merged**